src/sensai/data_transformation/dft.py

```diff
@@ -491,110 +491,162 @@ class DFTNormalisation(DataFrameTransformer):
     DFTNormalisation ignores N/A values during fitting and application.
     """
 
-    class RuleTemplate:
+    class _RuleBase:
         def __init__(self,
-                skip=False,
-                unsupported=False,
+                skip: bool = False,
+                unsupported: bool = False,
                 transformer: Optional[SkLearnTransformerProtocol] = None,
                 transformer_factory: Callable[[], SkLearnTransformerProtocol] = None,
-                independent_columns: Optional[bool] = None):
-            """
-            Creates a rule template which applies to one or more features/columns (depending on context).
-            Use parameters as follows:
+                independent_columns: Optional[bool] = None,
+                array_valued: bool = False,
+                fit: bool = True):
+            if (skip or unsupported) and count_not_none(transformer, transformer_factory) > 0:
+                raise ValueError("Passed transformer or transformer_factory while skip=True or unsupported=True")
+            self.skip = skip
+            self.unsupported = unsupported
+            self.transformer = transformer
+            self.transformerFactory = transformer_factory
+            self.independentColumns = independent_columns
+            self.arrayValued = array_valued
+            self.fit = fit
 
-                * If the relevant features are already normalised, pass ``skip=True``
-                * If the relevant features cannot be normalised (e.g. because they are categorical), pass ``unsupported=True``
-                * If the relevant features shall be normalised, the other parameters apply.
-                  No parameters, i.e. ``RuleTemplate()``, are an option if ...
 
-                    * a default transformer factory is specified in the :class:`DFTNormalisation` instance and its application
-                      is suitable for the relevant set of features.
-                      Otherwise, specify either ``transformerFactory`` or ``transformer``.
-                    * all relevant features are to be normalised in the same way.
-                      Otherwise, specify ``independentColumns=True``.
+    class RuleTemplate(_RuleBase):
+        """
+        A rule template from which a rule can be created by specifying a regex for the columns to which the rule applies.
```

Comment:
This is misleading, because the user might think that he needs to specify the regex.

I would suggest something like this:

A template from which a rule which matches multiple columns can be created.
This is useful for the generation of rules which shall apply to all the (numerical) columns generated by a feature generator (without specifically naming them).

Reply:
Done